lti_consumer/models.py

```diff
@@ -266,30 +266,38 @@ def _get_lti_1p3_consumer(self):
             )
 
             # Check if enabled and setup LTI-AGS
-            if self.block.has_score:
-
-                default_values = {
-                    'resource_id': self.block.location,
-                    'score_maximum': self.block.weight,
-                    'label': self.block.display_name,
-                }
-
-                if hasattr(self.block, 'start'):
-                    default_values['start_date_time'] = self.block.start
-
-                if hasattr(self.block, 'due'):
-                    default_values['end_date_time'] = self.block.due
-
-                # create LineItem if there is none for current lti configuration
-                lineitem, _ = LtiAgsLineItem.objects.get_or_create(
-                    lti_configuration=self,
-                    resource_link_id=self.block.location,
-                    defaults=default_values
-                )
+            if self.block.lti_advantage_ags_mode != 'disabled':
+                lineitem = None
+                # If using the declarative approach, we should create a LineItem if it
+                # doesn't exist. This is because on this mode the tool is not able to create
+                # and manage lineitems using the AGS endpoints.
+                if self.block.lti_advantage_ags_mode == 'declarative':
+                    # Set grade attributes
+                    default_values = {
+                        'resource_id': self.block.location,
+                        'score_maximum': self.block.weight,
+                        'label': self.block.display_name,
+                    }
+
+                    if hasattr(self.block, 'start'):
+                        default_values['start_date_time'] = self.block.start
+
+                    if hasattr(self.block, 'due'):
+                        default_values['end_date_time'] = self.block.due
+
+                    # create LineItem if there is none for current lti configuration
+                    lineitem, _ = LtiAgsLineItem.objects.get_or_create(
```

Comment:
Since the default value is `declarative` this will always create a line item when gets created. Are we deleting this default one once we change the xblock to use `programmatic` approach?

Reply:
@shimulch When using the programmatic approach, the tools can manage `LineItems` on their own, so they are free to delete or update the one the platform created. Additionally, the spec suggests sending one pre-created `LineItem` to serve as a straightforward migration path from the old `BasicOutcomesService`.